src/board/mod.rs

```diff
@@ -243,19 +257,90 @@ impl Board {
             let color = self.next_player();
             self.vacant
                 .iter()
-                .map(|coord| Play(color.clone(), coord.col, coord.row))
+                .map(|coord| Play(color, coord.col, coord.row))
                 .filter(|m| self.is_legal(*m).is_ok())
                 .collect()
         }
     }
-
```

Comment:
I tried to put the code directly into the function, but I got a borrowing error about borrowing board. I just put the code into board directly and for some reason it worked. The code definitely needs cleanup, though.